File: server/schemas/analytics.py
from pydantic import BaseModel


class ActivityEvent(BaseModel):
    type: str
    text: str
    at: str


class ActiveListingPreview(BaseModel):
    id: str
    name: str
    image_url: str | None = None
    price: float | None = None


class OverviewOut(BaseModel):
    active_listings: int
    pending_review: int
    total_earnings: float
    recent_activity: list[ActivityEvent]
    active_listings_preview: list[ActiveListingPreview]


class EarningsPoint(BaseModel):
    period: str
    amount: float


class BidsPoint(BaseModel):
    period: str
    count: int


class ListingPerformanceRow(BaseModel):
    listing_id: str
    product_name: str
    views: int
    watchlisted: int
    bids: int
    conversion_rate: float
    final_price: float | None = None
    vs_market_pct: float | None = None


class SellerAnalyticsOut(BaseModel):
    total_earnings: float
    bids_received: int
    listing_views: int
    watchlisted: int
    earnings_over_time: list[EarningsPoint]
    bids_over_time: list[BidsPoint]
    listing_performance: list[ListingPerformanceRow]


class WatchToggleOut(BaseModel):
    is_watching: bool
    watch_count: int

File: client/src/types/analytics.ts
export type ActivityEvent = {
  type: string;
  text: string;
  at: string;
};

export type ActiveListingPreview = {
  id: string;
  name: string;
  image_url: string | null;
  price: number | null;
};

export type Overview = {
  active_listings: number;
  pending_review: number;
  total_earnings: number;
  recent_activity: ActivityEvent[];
  active_listings_preview: ActiveListingPreview[];
};

export type EarningsPoint = {
  period: string;
  amount: number;
};

export type BidsPoint = {
  period: string;
  count: number;
};

export type ListingPerformanceRow = {
  listing_id: string;
  product_name: string;
  views: number;
  watchlisted: number;
  bids: number;
  conversion_rate: number;
  final_price: number | null;
  vs_market_pct: number | null;
};

export type SellerAnalytics = {
  total_earnings: number;
  bids_received: number;
  listing_views: number;
  watchlisted: number;
  earnings_over_time: EarningsPoint[];
  bids_over_time: BidsPoint[];
  listing_performance: ListingPerformanceRow[];
};
